src/gui/guiFormSpecMenu.cpp

```diff
@@ -556,23 +561,96 @@ void GUIFormSpecMenu::parseScrollBar(parserData* data, const std::string &elemen
 
 		spec.ftype = f_ScrollBar;
 		spec.send  = true;
-		gui::IGUIScrollBar* e =
-				Environment->addScrollBar(is_horizontal,rect,this,spec.fid);
+		GUIScrollBar *e = new GUIScrollBar(Environment, this, spec.fid, rect,
+				is_horizontal, true);
 
 		auto style = getStyleForElement("scrollbar", name);
 		e->setNotClipped(style.getBool(StyleSpec::NOCLIP, false));
+		e->setArrowsVisible(data->scrollBarOptions.showArrows);
+
+		s32 max = data->scrollBarOptions.max;
+		s32 min = data->scrollBarOptions.min;
+
+		e->setMax(max);
+		e->setMin(min);
 
-		e->setMax(1000);
-		e->setMin(0);
 		e->setPos(stoi(parts[4]));
-		e->setSmallStep(10);
-		e->setLargeStep(100);
+
+		e->setSmallStep(data->scrollBarOptions.smallStep);
+		e->setLargeStep(data->scrollBarOptions.largeStep);
+
+		s32 scrollbar_size;
+		if (is_horizontal)
+			scrollbar_size = dim.X;
+		else
+			scrollbar_size = dim.Y;
+
+		e->setPageSize(scrollbar_size * (max-min+1) / data->scrollBarOptions.thumbSize);
 
 		m_scrollbars.emplace_back(spec,e);
 		m_fields.push_back(spec);
 		return;
 	}
-	errorstream<< "Invalid scrollbar element(" << parts.size() << "): '" << element << "'"  << std::endl;
+	errorstream << "Invalid scrollbar element(" << parts.size() << "): '" << element
+		<< "'" << std::endl;
+}
+
+void GUIFormSpecMenu::parseScrollBarOptions(parserData* data, const std::string &element)
+{
+	std::vector<std::string> parts = split(element, ';');
```

Comment:
Would the following be possible?
```suggestion
	const std::vector<std::string> &parts = split(element, ';');
```